feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,537 @@
+    high cardinality. For categorical variables differences in distribution can be
+    investigated using a Chi-square or Kolmogorov-Smirnov test.
+
+    Parameters
+    ----------
+
+    split_col: string.
+        Label of the column according to which the dataframe will be split.
+
+    split_frac: float.
+        Ratio of the observations (when split_distinct is not activated) that goes
+        into the sub-dataframe that is used to determine the reference for the feature
+        distributions. The second sub-dataframe will be used to compare its feature
+        distributions to the reference ones.
+
+    split_distinct_values: boolean.
+        If set on, the split fraction does not account for the number of observations
+        but only for the number of distinct values in split_col.
+
+    variables: list, default=None
+        The list of variables to evaluate. If None, the transformer will evaluate all
+        numerical variables in the dataset.
+
+    switch_basis: boolean, default=False.
+        If set to true the role of the two matrices involved in the PSI calculations
+        (basis and measurement) will be switch. This is an important option as the
+        PSI value is not symmetric (i.e. PSI(a, b) != PSI(b, a)).
+
+    threshold: float, default = 0.25.
+        Threshold above which the distribution of a feature has changed so much that
+        the feature will be dropped. The most common values are 0.25 (large shift)
+        and 0.10 (medium shift).
+
+    strategy: string or callable, default='equal_frequency'
+        Type of binning used to represent the distribution of the feature. In can be
+        either "equal_width" for equally spaced bins or "equal_frequency" for bins
+        based on quantiles.
+
+    bins, int, default = 10
+        Number of bins used in the binning. For numerical feature a value of 10 is
+        considered as appropriate. For features with lower cardinality lower values
+        are usually used.
+
+    min_pct_empty_buckets, float, default = 0.0001
+        Value to add to empty bucket (when considering percentages). If a bin is
+        empty the PSI value may jump to infinity. By adding a small number to empty
+        bins, this issue is avoided. If the value added is too large, it may disturb
+        the calculations.
+
+    missing_values: str, default=ignore
+        Takes values 'raise' and 'ignore'. Whether the missing values should be raised
+        as error or ignored when determining correlation.
+
+    Attributes
+    ----------
+    features_to_drop_:
+        Set with the correlated features that will be dropped.
+
+    correlated_feature_sets_:
+        Groups of correlated features. Each list is a group of correlated features.
+
+    variables_:
+        The variables to consider for the feature selection.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    psi:
```

Comment:
all attributes learned during fit need to have a trailing underscore, to follow sklearn conventions, so psi_